src/overlays/actors/ovl_En_Bom_Chu/z_en_bom_chu.c

```diff
@@ -32,50 +37,539 @@ const ActorInit En_Bom_Chu_InitVars = {
+        lineLength *= 3.0f;
+        Math_Vec3f_Copy(&posA, &posB);
+
+        for (i = 0; i < 3; i++) {
+            if (i == 0) {
+                // backwards
+                posB.x = posA.x - (this->axisForwards.x * lineLength);
+                posB.y = posA.y - (this->axisForwards.y * lineLength);
+                posB.z = posA.z - (this->axisForwards.z * lineLength);
+            } else if (i == 1) {
+                // left
+                posB.x = posA.x + (this->axisLeft.x * lineLength);
+                posB.y = posA.y + (this->axisLeft.y * lineLength);
+                posB.z = posA.z + (this->axisLeft.z * lineLength);
+            } else {
+                // right
+                posB.x = posA.x - (this->axisLeft.x * lineLength);
+                posB.y = posA.y - (this->axisLeft.y * lineLength);
+                posB.z = posA.z - (this->axisLeft.z * lineLength);
+            }
+
+            if (EnBomChu_IsOnCollisionPoly(globalCtx, &posA, &posB, &posSide, &polySide, &bgIdSide)) {
+                isFloorPolyValid = EnBomChu_UpdateFloorPoly(this, polySide, globalCtx);
+                Math_Vec3f_Copy(&this->actor.world.pos, &posSide);
+                this->actor.floorBgId = bgIdSide;
+                break;
+            }
+        }
+
+        if (i == 3) {
+            // no collision nearby
+            EnBomChu_Explode(this, globalCtx);
+        }
+    }
+
+    if (isFloorPolyValid) {
+        EnBomChu_UpdateRotation(this);
+        this->actor.shape.rot.x = -this->actor.world.rot.x;
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        this->actor.shape.rot.z = this->actor.world.rot.z;
+    }
+
+    if (this->isMoving) {
+        func_800B8F98(&this->actor, NA_SE_IT_BOMBCHU_MOVE - SFX_FLAG);
+    }
+
+    if (this->actor.speedXZ != 0.0f) {
+        this->movingSpeed = this->actor.speedXZ;
+    }
+}
+
+void EnBomChu_Explode(EnBomChu* this, GlobalContext* globalCtx) {
+    EnBom* bomb;
+    s32 i;
+
+    bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x,
+                               this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0);
+
+    this->shouldTimerCountDown = true;
+    this->isMoving = false;
+
+    if (bomb != NULL) {
+        bomb->timer = 0;
+    }
+
+    this->timer = 1;
+    this->actor.speedXZ = 0.0f;
+
+    if (this->actor.depthInWater > 0.0f) {
+        for (i = 0; i < 40; i++) {
+            EffectSsBubble_Spawn(globalCtx, &this->actor.world.pos, 1.0f, 5.0f, 30.0f, 0.25f);
+        }
+    }
+
+    this->actor.draw = NULL;
+    this->actionFunc = EnBomChu_WaitForDeath;
+}
+
+void EnBomChu_WaitForDeath(EnBomChu* this, GlobalContext* globalCtx) {
+    if (this->timer == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+/**
+ * Transform coordinates from model space to world space, according to current orientation.
+ * `posModel` is expected to already be at world scale (1/100 compared to model scale)
+ */
```

Comment:
This isn't really correct, since model coordinates have the scale factor in them. It's more like the actor's special coordinates (z forward, x left, y up) to world. I think we settled on "actor coordinates" in the z_actor docs?

Reply:
Yeah I copied this straight from OoT without really understanding it much. Do you think it's still worthwhile to change the name of this when it will make it different from OoT?